onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -239,6 +239,26 @@ static Status GetCapabilityForEP(const GetCapabilityForEPParams& params) {
 }
 
 #if !defined(ORT_MINIMAL_BUILD)
+
+// This function queries the capabilities for a given EP, but it does not assign the nodes.
+// It also does not perform layout transformation. This will be done during normal partitioning.
+static Status GetCapabilityForEPForAotInlining(const Graph& graph,
+                                               const KernelRegistryManager& kernel_registry_mgr,
+                                               const IExecutionProvider& current_ep,
+                                               std::vector<std::unique_ptr<ComputeCapability>>& capabilities) {
+  const auto& ep_type = current_ep.Type();
+
+  const auto kernel_registries_for_ep = kernel_registry_mgr.GetKernelRegistriesByProviderType(ep_type);
+  const KernelLookup kernel_lookup{ep_type,
+                                   kernel_registries_for_ep,
+                                   kernel_registry_mgr.GetKernelTypeStrResolver()};
+
+  const GraphViewer graph_viewer(graph);
```

Comment:
Yes, we can. Would it be Okay if this does not match what Parition() does?

Reply:
I am not sure how exactly this would be done. Because we would need to connect these nodes somehow into a graph, and we don't know the partitioning logic. I think a better option would be to expose a new method that EPs can override, if they want a more efficient implementation, with the default being potentially a bit slower (like the current version). Eg., in practice, I suspect EPs can go a long way using just a set of (domain name, function name) pairs representing functions that they do NOT want to be inlined.

Reply:
`IndexedSubgraph` that acts as a filter requires graph metadata construction for filtering even with a single node (contrary to the comments in the header) so this becomes a bit expensive, but we could reuse the same graph viewer to query all of the eps.